Sources/MapboxNavigation/NavigationMapView.swift

```diff
@@ -549,7 +549,7 @@ open class NavigationMapView: UIView {
                     }
                 }
                 
-                try mapView.mapboxMap.style.addLayer(lineLayer, layerPosition: layerPosition)
+                try mapView.mapboxMap.style.addPersistentLayer(lineLayer, layerPosition: layerPosition)
```

Comment:
Do we need to make any changes to layers handling in `NavigationViewController`?

Reply:
In case is layer is persistent as I understand there is no pressing need to call `showRouteIfNeeded()`, but might be worth checking.

Reply:
I don't think so. Because during the active navigation, we do have the `RouteOverlayController` and `CarPlayNavigationViewController` that listen to the `.styleLoaded` change, and it would automatically update the layers.

Reply:
Should `addArrow(route:legIndexstepIndex:)` also use persistent layers?